enhancements/hypershift/disaster-recovery/hcp-etcd-backup-api.md

```diff
@@ -0,0 +1,587 @@
+
+- [ETCD Operator Backup API Design Doc](https://docs.google.com/document/d/1xBnK3D7MCblFgxQ8RUZFyWNIU_bnSFHciLhhHAeP0Xk/edit)
+- [ETCD Backup API implementation](https://github.com/openshift/api/pull/1482)
+- [Periodic Backup Controller](https://github.com/openshift/cluster-etcd-operator/pull/1074)
+
+The other field it's called `storeEndpoint` which basically will take care of the export of the binary to an external endpoint. This endpoint could be S3, Azure, etc.... Inside of this `storeEndpoint` we have some fields:
+
+- **BackupPrefix:** Where the user could set some kind of prefix to identify that backup in a more readable way. This Prefix will be parsed, validated and used in the final name which will contain the **timestamp**, the **HostedCluster name** and the provided **BackupPrefix**.
+- **Credentials:** This is a `SecretReference` located in the HostedCluster namespace using the same name as the backup plus the "credentials" suffix
+- **ServiceConfig:** This ServiceConfig field is another struct which it will contain the details about the remote destination you will upload that backup.
+
+The `ServiceConfig` struct will be the same on each provider. The plan is to provide a common interface and modify the behaviour depending on the destination provider.
+
+- **ServiceURL:** Points to the URL necessary to upload the binary.
+- **Region:** This will set the Region for the destination bucket.
+- **Tags:** Necessary tags to label the backup in he destination resources.
+
+### Workflow Description
+
+#### Backup
+
+When a user with a HostedCluster intends to perform a backup, the initial step involves modifying the HostedCluster by introducing a new field within the etcd API group. This new field will serve to incorporate the required parameters and configurations essential for executing backup operations seamlessly within the HostedCluster environment.
+
+```yaml
+...
+spec:
+  controllerAvailabilityPolicy: SingleReplica
+  dns:
+    baseDomain: hypershiftbm.lab
+  etcd:
+    managed:
+      storage:
+        persistentVolume:
+          size: 8Gi
+        restoreSnapshotURL: null
+        type: PersistentVolume
+    managementType: Managed
+    backup:
+      etcdBackup:
+        timeZone: "UTC"
+        pvcName: "pre-upgrade-backup"
+        retentionPolicy:
+          retentionType: RetentionNumber
+          retentionNumber:
+            maxNumberOfBackups: 10
+      storeEndpoint:
+        backupPrefix: "pre-upgrade"
+        credentials:
+          name: "s3-backup-credentials"
+        serviceConfig:
+          serviceURL: "s3://hc-sample-us-east-1"
+          region: "us-east"
+          tags:
+          - "hostedcluster=hc-sample"
+          - "foo=bar"
```

Comment:
this should be a map
```suggestion
          - hostedcluster: hc-sample
          - foo: bar
```

Reply:
We should avoid using maps in the API, but definitely should have a structure (see pod.spec.env)

Reply:
I was thinking in a slice with key=value, not a map actually.